### DISCUSSION TITLE
Unexpected stop of structure generation of molecular crystal

### DISCUSSION THREAD

**Discussion by takuyhaa** [Tomoki-YAMASHITA/CrySPY#59] [Q&A]
Thank you for developing a useful CSP tool.
I have a phenomenon that the structure generation of molecular crystals stops in the middle of the process without error.

cryspy.in
```
[basic]
algo = RS
calc_code = ASE
tot_struc = 1500
nstage = 1
njob = 10
jobcmd = zsh
jobfile = job_cryspy

[structure]
struc_mode = mol
natot =　192
atype = C S O N H 
nat = 104 4 4 4 76 
mol_file = ../../xyz/conformer_0.xyz
nmol = 4
spgnum = all

[ASE]
ase_python = ase_in.py

[option]
```

Terminal (last few lines)
```
[2023-12-26 09:10:39,256][gen_pyxtal][WARNING] Molecular symmetry is compatible with WP site
[H19 C26 S1 N1 O1]: C1: spg = 97 retry.
[2023-12-26 09:10:39,549][gen_pyxtal][WARNING] Molecular symmetry is compatible with WP site
[H19 C26 S1 N1 O1]: C1: spg = 131 retry.
[2023-12-26 09:10:39,850][gen_pyxtal][WARNING] Molecular symmetry is compatible with WP site
[H19 C26 S1 N1 O1]: C1: spg = 178 retry.

```

log_cryspy (last few lines under multi-processing)
```
[2023-12-26 09:09:40,838][gen_pyxtal][INFO] Structure ID    252 was generated. Space group:  12 -->  12 C2/m
[2023-12-26 09:09:40,858][gen_pyxtal][INFO] Structure ID   1075 was generated. Space group:  15 -->  15 C2/c
[2023-12-26 09:10:38,978][gen_pyxtal][INFO] Structure ID   1360 was generated. Space group:   1 -->   1 P1
```

After this last structure, no new structure is generated (no output in the terminal either). When other molecules are inputted at nmol = 4, I also sometimes encountered the same phenomenon. In this case, even if tot_struc is lowered (e.g. 100), structure generation did not complete. It seems that the structure generation stop at about 30 to 50% of tot_struc.
I confirmed that the same molecule completed tot_struc = 1500 with nmol = 2 without any problem.
Is this behavior a problem with cryspy or pyxtal? Is it a problem with my setting?

**Comment by Tomoki-YAMASHITA** [comment]
Sometimes the program stops when generating molecular crystals in pyxtal.
So timeout function is attached to CrySPY when struc_mode = mol.
The default value of timeout_mol is 120 s (https://tomoki-yamashita.github.io/CrySPY_doc/input/structure_sec/index.html).
If pyxtal crashes, it will move on to the next structure generation after 2 minutes.

If we leave it for a while and no progress is made, it might really be stuck somewhere, but we do not know the cause now.
I've never tried anything with that many atoms.

**Reply by takuyhaa** [reply]
Thank you for the reply.  
I checked the CPU usage during the multi-processing and found that the CPU usage gradually decreased even while the execution continued. 
This could be why structure generation did not complete at nmol=4.  
Therefore, I think the cause may lie in PyXtal, multi-processing, or a CPU memory limit.
I will investigate this phenomenon further.

**Reply by Tomoki-YAMASHITA** [reply]
If you need any help, let me know and I'll look into it.
I would appreciate it if you could let me know when something comes to light.
